src/intfuncs.c

```diff
@@ -26,6 +27,159 @@
 #include "stringobj.h"
 
 
+/****************************************************************************
+**
+** * * * * * * * "PCG" random numbers  * * * * * * * * * * * * *
+**
+** This next function implements the stepping function of the 'PCG'
+** random number library. This library produces high quality random numbers
+** which pass many statistical tests. They also have a very small state which
+** can be quickly initalised
+**
+** The following three functions are under the following original license
+** *Really* minimal PCG32 code / (c) 2014 M.E. O'Neill / pcg-random.org
+** Licensed under Apache License 2.0 (NO WARRANTY, etc. see website)
```

Comment:
I really like http://www.pcg-random.org , saw it before. On a technical level, I have no issues with this PR.

However, I note that the Apache License 2.0 is deemed incompatible with GPLv2 by the FSF and others. Of course GAP is licenses under "GPL 2 or later", and indeed, the Apache License 2.0 is compatible with GPLv3. But this change then kinda would make GAP "GPLv3" only "through the backdoor", which seems unfortunate. It'd be different if we changed the license explicitly. But then there are also reasons to dislike GPLv3 compared to GPLv2... *sigh*

Anyway, of course we could also ignore this. But I am pretty sure there are people (e.g. on Debian) who'd *really* dislike this.

Reply:
I'm going to email the author and see if they would be happy to dual license the code GPL v2. Have marked do not merge

Reply:
Any news on this?